backend/elasticsearch/src/main/java/org/hibernate/search/backend/elasticsearch/gson/impl/AbstractTypingJsonAccessor.java

```diff
@@ -56,6 +57,18 @@ private T fromElement(JsonElement parent) {
 			return null;
 		}
 		else if ( !expectedType.isInstance( parent ) ) {
+			if ( JsonElementTypes.ARRAY.isInstance( parent ) ) {
+				// Unexpected multi-valued item
+				JsonArray parentAsArray = JsonElementTypes.ARRAY.fromElement( parent );
+				// Let it go if there's zero or one item
```

Comment:
Why do not we raise an exception?

Reply:
Because in some cases, Elasticsearch returns an empty or one-element array when a field is single-valued. That's because Elasticsearch does not have any information about the cardinality of fields in its metamodel.